client/sites/tools/logs/index.tsx

```diff
@@ -251,3 +263,257 @@ export const SiteLogs = ( {
+		{ id: 'timestamp', label: 'Timestamp', enableSorting: false },
+		{ id: 'body_bytes_sent', label: 'Body bytes sent', enableSorting: false },
+		{ id: 'cached', label: 'Cached', enableSorting: false },
+		{ id: 'http_host', label: 'HTTP Host', enableSorting: false },
+		{ id: 'http_referer', label: 'Referrer', enableSorting: false },
+	];
+};
+
+const getVisibleFieldsForLogType = ( logType: LogType ) => {
+	if ( logType === 'php' ) {
+		return [ 'severity', 'timestamp', 'message' ];
+	}
+	return [ 'request_type', 'date', 'status', 'request_url' ];
+};
+
+const getSortFieldForLogType = ( logType: LogType ) => ( logType === 'php' ? 'timestamp' : 'date' );
+
+export const SiteLogsDataViews = ( { logType }: { logType: LogType } ) => {
+	// TODO:
+	// - DataViews:
+	//   - filters
+	//   - search
+	//   - max width for long cells
+	//   - spacing left/right
+	// - Fields:
+	//   - timestamp for web logs
+	// - Address the "show more" interaction.
+
+	const { __ } = useI18n(); // Can we use translate instead?
+
+	const moment = useLocalizedMoment();
+	const getLatestDateRange = useCallback( () => {
+		const startTime = moment().subtract( 7, 'd' );
+		const endTime = moment();
+		return { startTime, endTime };
+	}, [ moment ] );
+
+	const [ dateRange, setDateRange ] = useState( () => {
+		const latest = getLatestDateRange();
+		const dateRangeQuery = getDateRangeQueryParam( moment );
+		return {
+			startTime: dateRangeQuery.startTime || latest.startTime,
+			endTime: dateRangeQuery.endTime || latest.endTime,
+		};
+	} );
+	const handleTimeRangeChange = ( newStart: Moment | null, newEnd: Moment | null ) => {
+		if (
+			( ! newStart && ! newEnd ) ||
+			( dateRange.startTime.isSame( newStart ) && dateRange.endTime.isSame( newEnd ) )
+		) {
+			return;
+		}
+
+		// setIsMobileOpen( false ); // TODO
+		let startTime = newStart || dateRange.startTime;
+		let endTime = newEnd || dateRange.endTime;
+		if ( ! startTime.isValid() || ! endTime.isValid() ) {
+			const latest = getLatestDateRange();
+			startTime = latest.startTime;
+			endTime = latest.endTime;
+		}
+
+		setDateRange( { startTime, endTime } );
+		// setAutoRefresh( false ); // TODO
+		updateDateRangeQueryParam( { startTime, endTime } );
+	};
+
+	const [ view, setView ] = useState( () => {
+		return {
+			type: 'table' as const,
+			page: 1,
+			perPage: 50,
+			sort: {
+				field: getSortFieldForLogType( logType ),
+				direction: 'desc',
+			},
+			fields: getVisibleFieldsForLogType( logType ),
+		};
+	} );
+	const fields = useFields( { logType } );
+	const { data, paginationInfo, isLoading } = useDataLogs( { view, logType, dateRange } );
+	const onChangeView = ( newView: View ) =>
+		setView( ( oldView ) => ( {
+			...oldView,
+			...newView,
+			type: 'table' as const,
+		} ) );
+	useEffect( () => {
+		setView( ( view ) => ( {
+			...view,
+			sort: {
+				field: getSortFieldForLogType( logType ),
+			},
+			fields: getVisibleFieldsForLogType( logType ),
+		} ) );
+	}, [ logType ] );
```

Comment:
From what I see here, having `view` as a state is making your life harder. Maybe make all its dependencies states but the view itself can be a constructed in the end before returning.

```ts
const view = {
   type: 'table' as const,
   sort: { field: getSortFieldForLogType( logType ) },
   fields: getVisibleFieldsForLogType( logType ),
}
```